proto/pkg/chains/chains.proto

```diff
@@ -34,7 +35,57 @@ enum ChainName {
   amoy_testnet = 16;
 }
 
+// Network represents the network type of the chain
+enum Network {
+  option (gogoproto.goproto_enum_stringer) = true;
+  eth = 0;
+  zeta = 1;
+  btc = 2;
+  polygon = 3;
+  bsc = 4;
+}
+
+// NetworkType represents the network type of the chain
+enum NetworkType {
+  option (gogoproto.goproto_enum_stringer) = true;
+  mainnet = 0;
+  testnet = 1;
+  privnet = 2;
+  devnet = 3;
+}
+
+// Vm represents the virtual machine type of the chain to support smart contracts
+enum Vm {
+  option (gogoproto.goproto_enum_stringer) = true;
+  no_vm = 0;
+  evm = 1;
+}
+
+// Consensus represents the consensus algorithm used by the chain
+enum Consensus {
+  option (gogoproto.goproto_enum_stringer) = true;
+  ethereum = 0;
+  tendermint = 1;
+  bitcoin = 2;
+}
+
+//    Chain represents a blockchain network with its unique chain ID
+//    ChainName is the name of the chain
+//    ChainId is the unique identifier of the chain
+//    Network is the network type of the chain , this can be ZETA, ETH, BSC, BTC, POLYGON
+//    NetworkType is the network type of the chain, this can be MAINNET, TESTNET, DEVNET, PRIVNET
+//    Vm is the virtual machine type of the chain to support smart contracts, this can be EVM, NO_VM
+//    Consensus is the consensus algorithm used by the chain, this can be Tendermint, Ethereum, Bitcoin
+//    IsExternal is a boolean value to determine if the chain is external to Zeta
+//    IsHeaderSupported is a boolean value to determine if the chain supports headers
+
 message Chain {
   ChainName chain_name = 1;
   int64 chain_id = 2;
+  Network network = 3;
+  NetworkType network_type = 4;
+  Vm vm = 5;
+  Consensus consensus = 6;
+  bool is_external = 7;
+  bool is_header_supported = 8;
```

Comment:
I might think this one will not be necessary.

We would implement header system per Consensus from the consensus field, so we can instead have `IsHeaderSupported(Consensus) bool`

Then the flags in lightclient can choose which chain use headers/proofs

Reply:
>If we want to remove the is_header_supported flag , we should also modify the Verification flags to be using exact chain ids .

Yes, this is what we need to do. Because this structure should represent static information. This idea if a chain should be supported independenlty of the consensus is more something dynamic that can be handled with the flag.

Reply:
That would mean we need to store the information about which chain id's support block headers somewhere in the state.

Reply:
We should change the blockVerificationFlags object to hold a list of chainID->boolean

Reply:
https://github.com/zeta-chain/node/issues/2083